Add sections for variance errors

Masa-Shin opened the pull request.

Masa-Shin committed:
feat: add TS2637 section

Masa-Shin committed:
feat: add TS2636 section

bennycode commented on errors/includes/TS2xxx/TS2636.md:
```diff
@@ -0,0 +1,23 @@
+## TS2636
+
+> error TS2636: Type 'Func<super-T>' is not assignable to type 'Func<sub-T>' as implied by variance annotation. Type 'super-T' is not assignable to type 'sub-T'.
```
```suggestion
> error TS2636: Type '`Func<super-T>`' is not assignable to type '`Func<sub-T>`' as implied by variance annotation. Type '`super-T`' is not assignable to type '`sub-T`'.
```

bennycode reviewed (changes requested):
Hey, happy to see you again! 🙂 Let's put the variable values of an error message into code blocks so it's visible that this part of the error message is based on your custom code.

Masa-Shin committed:
put the variable values of an error message into code blocks.

https://github.com/bennycode/typescript-errors/pull/3#pullrequestreview-1399207429

Co-authored-by: Benny Neugebauer <mail@bennycode.com>

Masa-Shin commented:
Thanks! Updated.

bennycode commented:
Cool! It's live on https://typescript.tv/errors/#TS2636 - Thank you very much! 👍👍
